src/libs/location/src/Character.cpp

```diff
@@ -2450,6 +2477,103 @@ void Character::Update(float dltTime)
     {
         location->GetLights()->UpdateMovingLight(m_nHandLightID, GetHandLightPos());
     }
+
+    // Procedural head look
+
+    Animation *ani = m->GetAnimation();
+    if (!ani)
+        return;
+    
+    bool isControllableHead = ani->IsControllableHead();
+
+    if (curHeadLookState != headlook_default || isControllableHead)
+    {
+        const float headMovSpeed = 4.0f;
+
+        float destHeadAX = 0.0f;
+        float destHeadAY = 0.0f;
+
+        switch (curHeadLookState)
+        {
+        case headlook_default:
+            break;
+
+        case headlook_camera:
+            destHeadAX = camAng.x - 0.2f;
+            destHeadAY = atan2(sin(ay - camAng.y), cos(ay - camAng.y));
+            break;
+
+        case headlook_character:
+        case headlook_point: {
+            CVECTOR targetPos = 0.0f;
+            if (curHeadLookState == headlook_character)
+            {
+                auto *targetChrPtr = static_cast<Character *>(EntityManager::GetEntityPointer(headLookChrTarget));
+                if (targetChrPtr)
+                {
```

Comment:
nullptr means the entity was deleted. It would make sense to change `curHeadLookState` to `headlook_default` and break